### PR TITLE
Add htimedelta support.

### PR DESCRIPTION
**Pull request by rajnesh-kanwal**

### CODE REVIEW

File: riscv-regs/src/csrs/mod.rs
Comment (abrestic-rivos) [inline riscv-regs/src/csrs/mod.rs:40]: this is already available as `CSR.hpmcounter[1]`, no?

File: src/vm.rs
Comment (abrestic-rivos) [inline src/vm.rs:266]: I think we actually want to latch at the point the first vCPU in the TVM is run. If we latch at TVM creation, then the host can effectively control the start time the TVM observes (which could possibly effect, e.g., RNG seeds).

File: src/vm.rs
Comment (abrestic-rivos) [inline src/vm.rs:1211]: A possible simplification would be to not use `htimedelta` for the host VM, i.e. host VM time == real time. But this works too, so up to you.

File: riscv-regs/src/csrs/defs.rs
Comment (atulkharerivos) [inline riscv-regs/src/csrs/defs.rs:9]: Nit: Unrelated, but typo in commit message.

File: src/vm.rs
Comment (atishp04) [reply src/vm.rs:266]: So you want to do it in tvm_finalize ? That means the delta would not include the time to create a TVM (initial configuration/memory conversion/) which may be significant. Once we have a confidential page pool in place, conversion wouldn't be factor though.

File: src/vm.rs
Comment (atishp04) [reply src/vm.rs:1211]: That would be problem in migration of the hosts. No ?  
Disclaimer: I still think host in VS mode to actually be useful along with host in HS mode in the future.

File: src/vm_cpu.rs
Comment (atishp04) [inline src/vm_cpu.rs:411]: Instead of setting the htimedelta in VmCpuArchState during creation, define an helper function to set it ?
We don't know how migration APIs will look like right now but htimedelta needs to be updated at that time.
We may or may not need to create VmCpuArchState during that time. 

If you like to keep it this way, leave a TODO comment at least.

File: src/vm.rs
Comment (abrestic-rivos) [reply src/vm.rs:266]: Not `tvm_finalize` -- the first time the TVM is run, i.e. `tvm_vcpu_run(0)`. The intention is for the TVM to observe a start time of ~0.

File: src/vm.rs
Comment (abrestic-rivos) [reply src/vm.rs:1211]: Migrating the host VM sounds scary and I'm having a hard time thinking of a good use case for it :)

File: riscv-regs/src/csrs/mod.rs
Comment (rajnesh-kanwal) [reply riscv-regs/src/csrs/mod.rs:40]: Didn't notice that. Thanks :+1: 
I personally find it less confusing to have `time`, `cycle` and `insret` to be separate from `hpmcounter` csrs. I am leaving this as it is for now and revert to using `hpmcounter[1]`.

File: src/vm.rs
Comment (atishp04) [reply src/vm.rs:1211]: I don't think there is any use case right now. May be in the future. However, the cost of maintaining the htimedelta for the host VM is not much. IMO, It follows the standard convention of keeping htimedelta for any VM. Less confusion for future. 
Anyways, I am okay with your suggestion as well. Leave a good comment to explain the reasoning either way.

File: .project
Comment (abrestic-rivos) [inline .project:1]: Unintended `git add`?

File: src/vm.rs
Comment (abrestic-rivos) [inline src/vm.rs:514]: Does this need to be `pub`?

**Review by abrestic-rivos — COMMENTED** [review]
Two minor comments, otherwise LGTM.

File: .project
Comment (rajnesh-kanwal) [reply .project]: Oops. Fixed. Added it to gitignore as well.

File: src/vm.rs
Comment (rajnesh-kanwal) [reply src/vm.rs]: Fixed.

**Review by abrestic-rivos — APPROVED** [review]
LGTM. I'll let @atishp04 have another look before merging. Thanks!

**Review by atishp04 — APPROVED** [review]

**Comment by atishp04** [comment]
LGTM.

**Pull request merged**
